src/lib/parameters/CMakeLists.txt

```diff
@@ -169,7 +178,7 @@ if (NOT "${PX4_BOARD}" MATCHES "px4_io")
 
 	target_link_libraries(parameters PRIVATE perf tinybson px4_platform)
 
-	target_compile_definitions(parameters PRIVATE -DMODULE_NAME="parameters")
+	target_compile_definitions(parameters PRIVATE -DMODULE_NAME="parameters" -D__KERNEL__)
```

Comment:
FYI: the __KERNEL__ flag here is needed only for nuttx includes. Without this, the nuttx work queues are allocated wrongly. The flag is not used anywhere in posix build, and has not effect in flat build.